### ISSUE TITLE
Allow states simultaneously with death

### ISSUE THREAD

**Issue by fmatthew5876** [EasyRPG/Player#2327]
In player whenever we add death state we clear all other states. In addition, our battle algorithms are setup to abort early if death is added.

This behavior is incorrect. 

RPG_RT actually lets you have other states simultaneously with death. This is however uncommon, because death always has priority 100, and so any other state which is allowed to coexist must have priority 90 or higher.

Simply create another state with priority=100 and do this event code:
```
AddState: OtherState
AddState: Death
RemoveState:Death
```

The actor will still have OtherState

- [ ] Adding states ignores death
- [ ] Skills will abort all effects if the damage component kills the target

**Comment by rueter37** [comment]
I did some tests with states in RPG_RT and made the following observations:
- States must have a priority of at least 91 to not get removed by death
- States get only added/removed by skills and weapons if the target is not killed by damage (the weapons part contradicts with your second checkbox item)

I made a branch which fixes this issue but it depends on #2329. As soon as #2329 is merged into master I will open a PR with the fix.

**Comment by rueter37** [comment]
Added a PR (#2350) which should fix this issue.
